src/operator/nn/mkldnn/mkldnn_base-inl.h

```diff
@@ -277,6 +277,19 @@ inline static mkldnn::memory::desc GetWeightDesc(const NDArray &arr,
   }
 }
 
+inline static const std::vector<NDArray> GetMKLDNNInputArray(const std::vector<NDArray> &inputs) {
+  std::vector<NDArray> ret;
+  ret.reserve(inputs.size());
+  for (const auto &in : inputs) {
+    if (in.IsView() && in.IsMKLDNNData()) {
+      ret.push_back(in.Reorder2Default());
+    } else {
+      ret.push_back(in);
```

Comment:
Return `inputs` here to avoid overhead?

Reply:
pls let author @ZhennanQin help to check

Reply:
Yes, we can do that.

Reply:
done